control-plane/csi-driver/src/bin/node/dev/nvmf.rs

```diff
@@ -359,5 +371,19 @@ pub(crate) fn transport_from_url(url: &Url) -> Result<TrType, DeviceError> {
         .split('+')
         .nth(1)
         .unwrap_or(default_xprt.as_str());
-    TrType::from_str(xprt).map_err(|e| DeviceError::new(format!("{e:?}").as_str()))
+
+    let ret_xprt = TrType::from_str(xprt).map_err(|e| DeviceError::new(format!("{e:?}").as_str()));
+    let connect_cap_check = RDMA_CONNECT_CHECK.get().unwrap_or(&(false, false));
+
+    if !connect_cap_check.0 {
```

Comment:
I'm a bit confused with the bools, if we don't check, shouldn't we check if RDMA is supported? otherwise  might as well bail out?

Reply:
> I'm a bit confused with the bools, if we don't check, shouldn't we check if RDMA is supported? otherwise might as well bail out?

The logic is: If fallback is not enabled via cli, we return `ret_xprt` as-is. It could be tcp or rdma depending on nvmf url. If connect fails for rdma, let it fail.

Reply:
Do we need a good user message when we fail later on? Otherwise we can fail here with our own error message.

Reply:
let's not fail here.
If it fails due to incapability later on, the trace that we've in main_.rs:363 will be relevant.

Reply:
Can you please check which error message we log in this case?